build/scripts/build-libffi.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+function build {
+    local ARCH=$1
+    local BINARY_DIR="build_$PLATFORM_NAME-$ARCH-$CONFIGURATION"
+
+    if [ -f "$BINARY_DIR/.libs/libffi.a" ]; then
+        return 0
+    fi
+
+    local TRIPLE
+    if [ "$ARCH" == "i386" ]; then
+        TRIPLE="i386-apple-darwin11"
+    elif [ "$ARCH" == "x86_64" ]; then
+        TRIPLE="x86_64-apple-darwin13"
+    elif [ "$ARCH" == "armv7" ]; then
+        TRIPLE="arm-apple-darwin11"
+    elif [ "$ARCH" == "arm64" ]; then
+        TRIPLE="aarch64-apple-darwin13"
+    else
+        echo "Invalid architecture: $ARCH"
+        exit 1
+    fi
+
+    autoreconf -i
+
+    mkdir -p "$BINARY_DIR" && pushd "$_"
+
+    local COMPILER_FLAGS="-arch $ARCH -isysroot $SDKROOT -miphoneos-version-min=$IPHONEOS_DEPLOYMENT_TARGET"
```

Comment:
Taking as much variables from Xcode environment is a great idea :smile: 

I've also edited the hardcoded optimization levels and debugging symbols flags.